src/controller/CHIPDeviceController.cpp

```diff
@@ -1025,6 +1028,114 @@ CHIP_ERROR DeviceCommissioner::UnpairDevice(NodeId remoteDeviceId)
     return CHIP_NO_ERROR;
 }
 
+CHIP_ERROR DeviceCommissioner::SendEchoRequest(NodeId remoteDeviceId)
+{
+    CHIP_ERROR err = CHIP_NO_ERROR;
+
+    Messaging::SendFlags sendFlags;
+    System::PacketBufferHandle payloadBuf;
+    char * requestData = nullptr;
+
+    uint32_t size = mPingParams.GetEchoReqSize();
+    VerifyOrExit(size <= kMaxPayloadSize, err = CHIP_ERROR_MESSAGE_TOO_LONG);
+
+    requestData = static_cast<char *>(chip::Platform::MemoryAlloc(size));
+    VerifyOrExit(requestData != nullptr, err = CHIP_ERROR_NO_MEMORY);
+
+    snprintf(requestData, size, "Echo Message %" PRIu32 "\n", mPingParams.GetEchoReqCount() + 1);
+
+    payloadBuf = MessagePacketBuffer::NewWithData(requestData, size);
+    VerifyOrExit(!payloadBuf.IsNull(), err = CHIP_ERROR_NO_MEMORY);
+
+    if (mPingParams.IsUsingMRP())
+    {
+        sendFlags.Set(Messaging::SendMessageFlags::kNone);
+    }
+    else
+    {
+        sendFlags.Set(Messaging::SendMessageFlags::kNoAutoRequestAck);
+    }
+
+    ChipLogDetail(Controller, "Send echo request message with payload size: %d bytes to Node: %" PRIu64, size, remoteDeviceId);
```

Comment:
```suggestion
    ChipLogDetail(Controller, "Send echo request message with payload size: %" PRIu32 " bytes to Node: %" PRIu64, size, remoteDeviceId);
```

Reply:
And for the PRIu64, that's not supported on all platforms.  You may want ChipLogFormatX64/ChipLogValueX64